ui/app/manager/src/pages/page-account.ts

```diff
@@ -439,91 +450,140 @@ export class PageAccount extends Page<AppStateKeyed>  {
      * If the password has been changed, it will, after updating the user, also request to reset the password.
      */
     protected async _updateUser(user: UserModel): Promise<void> {
-        await manager.rest.api.UserResource.update(manager.getRealm(), user).then(() => {
-            if(user.password) {
+        try {
+            await manager.rest.api.UserResource.update(user);
+            
+            if (user.password) {
                 const credentials = {value: user.password} as Credential;
-                manager.rest.api.UserResource.resetPassword(manager.getRealm(), user.id, credentials)
-                    .then(() => {
-                        showSnackbar(undefined, "saveUserSucceeded");
-                        })
-                    .catch(() => {
-                        showSnackbar(undefined, "saveUserFailed");
-                    });
-                this._dirty = false;
+                try {
+                    await manager.rest.api.UserResource.resetPassword(credentials);
+                    showSnackbar(undefined, "saveUserSucceeded");
+                } catch (e) {
+                    showSnackbar(undefined, "saveUserFailed");
+                    return;
+                }
+            } else {
+                showSnackbar(undefined, "saveUserSucceeded");
             }
-        })
+            
+            // Update the stored user with the saved version and reset dirty flag
+            this._user = {...user};
+            this._dirty = false;
+            
+            // Force re-render to update UI elements (like the reset password button)
+            this.requestUpdate();
```

Comment:
Looking through the code, I don't see any reason that this `requestUpdate()` is necessary.
You're updating the variable in full, so the lifecycle hook is called correctly.
Let me know if I'm wrong.

Reply:
I haven't fully reviewed the UI code changes that Michael made